src/MoBi.Engine/Sbml/SBMLTask.cs

```diff
@@ -105,14 +105,7 @@ public Model GetModel(string filename)
          {
             throw new MoBiException(SBMLConstants.ModelNotRead(sbmlDoc.getErrorLog().ToString()));
          }
-         try
-         {
-            convertSBML(sbmlDoc);
-         }
-         catch (Exception _)
-         {
-            throw new MoBiException(SBMLConstants.NoValidConversionToLevel3Version2);
-         }
+         convertSBML(sbmlDoc);
```

Comment:
I don't get it. Now you do not catch anything at all? Don't you want to catch when you have the wrong version

Reply:
It does throw the exception. convertSBML is actually checking and throwing the exception. Here we are just catching everything and translating to no valid conversion when actually this is already done in line 137 where the return from setLevelAndVersion is been checked.